sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -877,6 +878,66 @@ case class GpuFromUnixTime(
   override lazy val resolved: Boolean = childrenResolved && checkInputDataTypes().isSuccess
 }
 
+
+class FromUTCTimestampExprMeta(
+    expr: FromUTCTimestamp,
+    override val conf: RapidsConf,
+    override val parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends BinaryExprMeta[FromUTCTimestamp](expr, conf, parent, rule) {
+
+  override def tagExprForGpu(): Unit = {
+    val timezoneShortID = extractStringLit(expr.right).getOrElse("")
+
+    if (timezoneShortID.isEmpty) {
+      willNotWorkOnGpu("input time zone is null or empty")
```

Comment:
Curious, what does the CPU do for this case?  Does it throw, use the system timezone, or something else?

Reply:
Oh the Spark CPU throw exception:
```
22/08/22 15:31:22 ERROR Executor: Exception in task 0.0 in stage 3.0 (TID 3)
java.time.DateTimeException: Invalid ID for ZoneOffset, invalid format: 
	at java.time.ZoneOffset.of(ZoneOffset.java:241)
	at java.time.ZoneId.of(ZoneId.java:405)
	at java.time.ZoneId.of(ZoneId.java:359)
	at java.time.ZoneId.of(ZoneId.java:315)
	at org.apache.spark.sql.catalyst.util.DateTimeUtils$.getZoneId(DateTimeUtils.scala:55)
	at org.apache.spark.sql.catalyst.util.DateTimeUtils.getZoneId(DateTimeUtils.scala)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.init(Unknown Source)
	at org.apache.spark.sql.execution.WholeStageCodegenExec.$anonfun$doExecute$4(WholeStageCodegenExec.scala:752)
	at org.apache.spark.sql.execution.WholeStageCodegenExec.$anonfun$doExecute$4$adapted(WholeStageCodegenExec.scala:749)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2(RDD.scala:915)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2$adapted(RDD.scala:915)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
However, if the input time zone is `null` then:
```
scala> val df = Seq(Timestamp.valueOf("1999-12-24 03:58:55.000826"), Timestamp.valueOf("2020-12-12 01:02:33.123456")).toDF("v")
df: org.apache.spark.sql.DataFrame = [v: timestamp]

scala> df.repartition(1).selectExpr("from_utc_timestamp(v, NULL)").show(false)
+-------------------------------------------+
|from_utc_timestamp(v, CAST(NULL AS STRING))|
+-------------------------------------------+
|null                                       |
|null                                       |
+-------------------------------------------+
```

Reply:
So maybe we'll need to run on GPU for the null case to generate an all-null output column.

Reply:
Yes, we need to handle a NULL timezone gracefully and return a NULL for every row in the batch, but we shouldn't check for an empty timezone ID and simply let it throw when trying to parse like Spark CPU does.